xdsl/dialects/stream.py

```diff
@@ -49,37 +47,93 @@ def __init__(self, element_type: _StreamTypeElement):
     def get_element_type(self) -> _StreamTypeElement:
         return self.element_type
 
+    @overload
+    @classmethod
```

Comment:
Having these as `@staticmethod`s should allow you to specify the generic type properly and avoid having to add `[Attribute]` below. Think a similar thing came up on the previous `constr` additions

Reply:
Yep I had the same thought just pushed. Forgot that that was the conclusion of the previous iteration's discussion.

Reply:
I might go through and fix this in other places

Reply:
Unless that will clash with your next steps

Reply:
The only other one is in stencil and it looks like it'd be a real pain to get rid of